src/Transports/MassTransit.ActiveMqTransport/ActiveMqTransport/Middleware/ActiveMqConsumerFilter.cs

```diff
@@ -110,6 +107,32 @@ async Task<ActiveMqConsumer> CreateConsumer(SessionContext context, Queue entity
         }
 
 
+        class ConsumerSupervisor :
+            Supervisor
+        {
+            public ConsumerSupervisor(ActiveMqConsumer[] consumers)
+            {
+                foreach (var consumer in consumers)
+                {
+                    consumer.Completed.ContinueWith(async _ =>
+                    {
+                        try
+                        {
+                            if (!IsStopping)
+                                await this.Stop("Consumer stopped, stopping supervisor").ConfigureAwait(false);
```

Comment:
so we're calling Stop multiple times? Or does Stop already check for this and simply return?

Reply:
Oh, I see, !IsStopping duh.